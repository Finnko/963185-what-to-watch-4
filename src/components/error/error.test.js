import React from 'react';
import renderer from 'react-test-renderer';
import {Router} from 'react-router-dom';
import {createMemoryHistory} from 'history';
import Error from './error.jsx';

const mock = `Something went wrong`;

describe(`Error component render correctly`, () => {
  it(`Should Error component render correctly`, () => {
    const history = createMemoryHistory();

    const tree = renderer
      .create(
          <Router history={history}>
            <Error error={mock}/>
          </Router>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
