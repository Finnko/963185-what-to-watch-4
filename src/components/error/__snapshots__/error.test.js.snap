// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error component render correctly Should Error component render correctly 1`] = `
Array [
  <header
    className="page-header "
  >
    <div
      className="logo"
    >
      <a
        className="logo__link"
        href="/"
        onClick={[Function]}
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <div
      className="user-block"
    >
      <a
        className="user-block__link"
        href="/login"
        onClick={[Function]}
      >
        Sign In
      </a>
    </div>
  </header>,
  <section
    className="catalog"
  >
    <div
      className="page-content"
    >
      <div
        className="error-wrapper"
      >
        <p
          className="error-message"
        >
          Something went wrong
        </p>
      </div>
    </div>
  </section>,
  <footer
    className="page-footer"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link logo__link--light"
        href="/"
        onClick={[Function]}
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <div
      className="copyright"
    >
      <p>
        Â© 2019 What to watch Ltd.
      </p>
    </div>
  </footer>,
]
`;
